cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/templates/etl/batch/BatchDriver.java

```diff
@@ -0,0 +1,206 @@
+    DefaultStageConfigurer sinkConfigurer = new DefaultStageConfigurer(sink.getClass());
+    sink.configure(sinkConfigurer);
+    BatchSinkContext sinkContext = new MapReduceSinkContext(context, sinkConfigurer.createSpecification());
+    sink.prepareJob(sinkContext);
+  }
+
+  private void prepareTransform(MapReduceContext context) {
+    Job job = context.getHadoopJob();
+    JsonObject config = JSON_PARSER.parse(context.getRuntimeArguments().get(Constants.CONFIG_KEY)).getAsJsonObject();
+    JsonArray transformArray = config.get(Constants.TRANSFORM_KEY).getAsJsonArray();
+
+    //Set list of Transform classes to be used in the Mapper
+    job.getConfiguration().set(Constants.Transform.TRANSFORM_CLASS_LIST,
+                               context.getRuntimeArguments().get(Constants.Transform.TRANSFORM_CLASS_LIST));
+
+    //Set the properties of the Transforms to be used in the initialization in the Mapper
+    job.getConfiguration().set(Constants.Transform.TRANSFORM_PROPERTIES, GSON.toJson(transformArray));
+  }
+
+  @Override
+  public void onFinish(boolean succeeded, MapReduceContext context) {
+    //TODO: Get the Adapter Name and include it in the log
+    LOG.info("BatchDriver Run {}", succeeded);
+  }
+
+  /**
+   * Mapper Driver for ETL Transforms.
+   */
+  public static class MapperDriver extends Mapper implements ProgramLifecycle<MapReduceContext> {
+    private static final Gson GSON = new Gson();
+    private static final JsonParser JSON_PARSER = new JsonParser();
+    private static final Type STRING_LIST_TYPE = new TypeToken<List<String>>() { }.getType();
+    private static final Type STRING_MAP_TYPE = new TypeToken<Map<String, String>>() { }.getType();
+    private TransformExecutor transformExecutor;
+
+    private MapReduceContext mapReduceContext;
+    private List<Transform> transforms = Lists.newArrayList();
+
+    @Override
+    public void initialize(MapReduceContext context) throws Exception {
+      this.mapReduceContext = context;
+    }
+
+    @Override
+    public void setup(Context context) throws IOException, InterruptedException {
+      super.setup(context);
+      //Get transform class names from the context.
+      String transformClassList = context.getConfiguration().get(Constants.Transform.TRANSFORM_CLASS_LIST);
+      List<String> classList = GSON.fromJson(transformClassList, STRING_LIST_TYPE);
+      JsonArray transformArray = JSON_PARSER.parse(context.getConfiguration().get(
+        Constants.Transform.TRANSFORM_PROPERTIES)).getAsJsonArray();
+      LOG.info("Transform Classes to be used : {}", classList);
+      LOG.info("Properties for Transform Classes : {}", transformArray);
+      Preconditions.checkArgument(classList.size() == transformArray.size());
+      instatiateTransforms(classList);
+      initializeTransforms(transforms, transformArray);
+      transformExecutor = new TransformExecutor(transforms);
+    }
+
+    private void instatiateTransforms(List<String> classList) {
```

Comment:
Why just a `String`? What if transform class need properties (e.g. some regex for filtering?). I think transform config should be an object.